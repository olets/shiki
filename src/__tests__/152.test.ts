// https://github.com/shikijs/shiki/issues/152
import { getHighlighter } from '../index'

test('Correctly highlights LaTeX', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['tex', 'latex']
  })

  const out = highlighter.codeToThemedTokens(`%\\usepackage{pkg}`, 'latex', 'nord', {
    includeExplanation: true
  })
  console.log(JSON.stringify(out, null, 2))
  expect(out).toMatchSnapshot()
})
