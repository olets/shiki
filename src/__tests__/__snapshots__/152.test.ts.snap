// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly highlights LaTeX 1`] = `
Array [
  Array [
    Object {
      "color": "#616E88",
      "content": "%\\\\usepackage{pkg}",
      "explanation": Array [
        Object {
          "content": "%",
          "scopes": Array [
            Object {
              "scopeName": "text.tex.latex",
              "themeMatches": Array [],
            },
            Object {
              "scopeName": "comment.line.percentage.tex",
              "themeMatches": Array [
                Object {
                  "name": "Comment",
                  "scope": "comment",
                  "settings": Object {
                    "foreground": "#616E88",
                  },
                },
              ],
            },
            Object {
              "scopeName": "punctuation.definition.comment.tex",
              "themeMatches": Array [
                Object {
                  "name": "Punctuation",
                  "scope": "punctuation",
                  "settings": Object {
                    "foreground": "#ECEFF4",
                  },
                },
                Object {
                  "name": "Punctuation Definition Comment",
                  "scope": Array [
                    "punctuation.definition.comment",
                    "punctuation.end.definition.comment",
                    "punctuation.start.definition.comment",
                  ],
                  "settings": Object {
                    "foreground": "#616E88",
                  },
                },
              ],
            },
          ],
        },
        Object {
          "content": "\\\\usepackage{pkg}",
          "scopes": Array [
            Object {
              "scopeName": "text.tex.latex",
              "themeMatches": Array [],
            },
            Object {
              "scopeName": "comment.line.percentage.tex",
              "themeMatches": Array [
                Object {
                  "name": "Comment",
                  "scope": "comment",
                  "settings": Object {
                    "foreground": "#616E88",
                  },
                },
              ],
            },
          ],
        },
      ],
      "fontStyle": 0,
    },
  ],
]
`;
